mmdet/apis/ote/apis/detection/openvino_task.py

```diff
@@ -182,3 +210,63 @@ def evaluate(self,
                  output_result_set: ResultSet,
                  evaluation_metric: Optional[str] = None):
         return MetricsHelper.compute_f_measure(output_result_set).get_performance()
+
+    def optimize(self,
+                 optimization_type: OptimizationType,
+                 dataset: Optional[Dataset],
+                 output_model: OptimizedModel,
+                 optimization_parameters: Optional[OptimizationParameters]):
+
+        model_name = self.hparams.algo_backend.model_name.replace(' ', '_')
+        if optimization_type is not OptimizationType.POT:
+            raise RuntimeError("POT is the only supported optimization type for OpenVino models")
+
+        data_loader = ScOpenVinoDataLoader(dataset, self.inferencer)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            xml_path = os.path.join(tempdir, model_name+".xml")
+            bin_path = os.path.join(tempdir, model_name+".bin")
+            open(xml_path, "wb").write(self.model.get_data("openvino.xml"))
+            open(bin_path, "wb").write(self.model.get_data("openvino.bin"))
+
+            model_config = ADDict({
+                'model_name': model_name,
+                'model': xml_path,
+                'weights': bin_path
+            })
+
+            model = load_model(model_config)
+
+        engine_config = ADDict({
+            'device': 'CPU'
+        })
+
+        stat_subset_size = self.hparams.pot_parameters.stat_subset_size
+        preset = self.hparams.pot_parameters.preset.name.lower()
+
+        algorithms = [
+            {
+                'name': 'DefaultQuantization',
+                'params': {
+                    'target_device': 'ANY',
+                    'preset': preset,
+                    'stat_subset_size': min(stat_subset_size, len(data_loader))
+                }
+            }
+        ]
+
+        engine = IEEngine(config=engine_config, data_loader=data_loader, metric=None)
+
+        pipeline = create_pipeline(algorithms, engine)
+
+        compressed_model = pipeline.run(model)
+
+        compress_model_weights(compressed_model)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            save_model(compressed_model, tempdir, model_name=model_name)
+            output_model.set_data("openvino.xml", open(os.path.join(tempdir, model_name+".xml"), "rb").read())
+            output_model.set_data("openvino.bin", open(os.path.join(tempdir, model_name+".bin"), "rb").read())
+
+        self.model = output_model
```

Comment:
Is it possible to optimize already optimized model?

Reply:
Please point me to the place, where you handle that kind of situations here.

Reply:
I didn't find such functionality inside POT API, it just failed inside POT when int8 ir is loaded and optimization is started

Reply:
I mean, you should forbid the user to so via OpenVINO Task API then.

Reply:
Added

Reply:
I'm not sure, that raising an error is the best way to go. It might cause troubles at the backend side, but this is just my guess. Anyway, I'd consider setting the corresponding status for the `output_model` and returning from the method.

Reply:
something like
```
            if get_nodes_by_type(model, ['FakeQuantize']):
                logger.info("Model is already optimized by POT")
                self.model.model_status = ModelStatus.NOT_IMPROVED
                return
```
?

Reply:
Yes, kind of. Or like this one:

```python
            if get_nodes_by_type(model, ['FakeQuantize']):
                logger.warning("Model is already optimized by POT")
                self.model.model_status = ModelStatus.FAILED
                return

```

Reply:
Ok, done